src/Test/Utilities/Portable/Diagnostics/OperationTestAnalyzer.cs

```diff
@@ -971,7 +971,13 @@ public sealed override void Initialize(AnalysisContext context)
                  (operationContext) =>
                  {
                      IInstanceReferenceExpression instanceReference = (IInstanceReferenceExpression)operationContext.Operation;
-                     operationContext.ReportDiagnostic(Diagnostic.Create(instanceReference.InstanceReferenceKind == InstanceReferenceKind.Implicit ? ImplicitInstanceDescriptor : ExplicitInstanceDescriptor, instanceReference.Syntax.GetLocation()));
+                     operationContext.ReportDiagnostic(Diagnostic.Create(instanceReference.Syntax.IsKind(CSharp.SyntaxKind.ThisExpression) ||
```

Comment:
>instanceReference.Syntax.IsKind(CSharp.SyntaxKind.ThisExpression) [](start = 73, length = 65)

Rather than doing syntax checks, could we rely on ```IsImplicit``` flag? Also, is the ```IsKind``` helper strongly typed, can it match syntax nodes from one language to kinds from another?